src/sideBar.ts

```diff
@@ -0,0 +1,737 @@
+        if (cmakeProject) {
+            BaseNode.updateActiveProject(cmakeProject);
+        }
+        await this.refresh();
+    }
+
+    public async refresh(node?: BaseNode): Promise<any> {
+        if (node) {
+            await node.refresh();
+            this._onDidChangeTreeData.fire(node);
+        } else {
+            this._onDidChangeTreeData.fire(undefined);
+        }
+    }
+
+    clear(): Promise<any> {
+        BaseNode.updateActiveProject(undefined);
+        return this.refresh();
+    }
+
+    dispose(): void {
+        Disposable.from(...this.disposables).dispose();
+    }
+
+    getTreeItem(node: BaseNode): TreeItem {
+        return node.getTreeItem();
+    }
+
+    async getChildren(node?: BaseNode | undefined): Promise<BaseNode[]> {
+        // Initializing the tree for the first time
+        if (!node) {
+            const configNode = new ConfigNode();
+            await configNode.initialize();
+            const buildNode = new BuildNode();
+            await buildNode.initialize();
+            const testNode = new TestNode();
+            await testNode.initialize();
+            const debugNode = new SingleItemNode(TargetType.Debug);
+            await debugNode.initialize();
+            const launchNode = new SingleItemNode(TargetType.Launch);
+            await launchNode.initialize();
+            const projectNode = new SingleItemNode(TargetType.Project);
+            await projectNode.initialize();
+            return [configNode, buildNode, testNode, debugNode, launchNode, projectNode];
+        } else {
+            return node.getChildren();
+        }
+    }
+
+}
+
+export class BaseNode extends TreeItem {
+
+    static sideBarTreeView?: TreeView<BaseNode>;
+    static cmakeProject?: CMakeProject;
+
+    static initializeTreeView(sideBarTreeView: TreeView<BaseNode>) {
+        BaseNode.sideBarTreeView = sideBarTreeView;
+    }
+
+    static updateActiveProject(cmakeProject?: CMakeProject): void {
+        // Use project to create the tree
+        if (cmakeProject) {
+            BaseNode.cmakeProject = cmakeProject;
+        }
+    }
+
+    getTreeItem(): TreeItem {
+        return this;
+    }
+
+    getChildren(): BaseNode[] {
+        return [];
+    }
+
+    async initialize(): Promise<void> {
+    }
+
+    async refresh(): Promise<void> {
+    }
+
+    changeToStop(): void {
+    }
+
+    changeBackToOriginal(): void {
+    }
+}
+
+export class ConfigNode extends BaseNode {
+
+    private kit?: KitNode;
+    private variant?: VariantNode;
+    private preset?: PresetNode;
+
+    constructor() {
+        super(NodeType.Configure);
+    }
+
+    async initialize(): Promise<void> {
+        if (!BaseNode.cmakeProject) {
+            return;
+        }
+        this.command = {
+            title: localize('Configure', 'Configure'),
+            command: 'cmake.sideBar.configure',
+            arguments: [this, BaseNode.cmakeProject.workspaceFolder]
+        };
+        this.tooltip = "Configure";
```

Comment:
Don't forget to localize tooltips. I think in most cases you can grab the command title which is already localized.